base/math.jl

```diff
@@ -980,6 +980,26 @@ function _exponent_finite_nonzero(x::T) where T<:IEEEFloat
     return k - exponent_bias(T)
 end
 
+function _ilog2_step(y::T, d::T, s) where {T<:Integer}
+    if fld(y, d) ≥ d
+        y, n = _ilog2_step(y, d*d, s+s)
+    else
+        n = 0
+    end
+    if y ≥ d
+        y = fld(y, d)
```

Comment:
is the compiler figurign out that d is a power of 2?

Reply:
I don't think so. but it wasn't clear to me what the contract of `Integer` is. `fld` seemed like a safer bet  generically than `>>`

Reply:
IMO, `>>` would be better. It should be as easy to support for integers (they can always implement it with `fld` if they want)